milestone-3/frontend/src/components/recipe_page.jsx

```diff
@@ -0,0 +1,129 @@
+import './recipe_page.css';
+import Navbar from './navbar.jsx';
+import React, { useState, useEffect, useRef} from 'react';
+
+
+
+function Recipe_page(prop) {
+
+  //states for the ingrediants and instructions in list
+  const [newIngredient, setNewIngredient] = React.useState('');
+  const [finalIngr, setFinalIngr] = React.useState('');
+  const [newInstruction, setNewInstruction] = React.useState('');
+  const [finalInst, setFinalInst] = React.useState('');
+
+  //functions that will add the input into the javaS object and then update the html
+  let addIngr = () =>{
+      prop.recipe.ingrediantList.push(newIngredient);
+      setFinalIngr(newIngredient)
+      setNewIngredient("");
+      console.log(newIngredient);
+  }
+
+  let addInst = () =>{
+    prop.recipe.instructionList.push(newInstruction);
+    setFinalInst(newInstruction)
+    setNewInstruction("");
+    console.log(newInstruction);
+  }
+
+  const firstUpdate1 = useRef(true);
+  useEffect(() => {
+    //stop the initial render of useEffect()
+    if (firstUpdate1.current) {
+      firstUpdate1.current = false;
+      return;
+    }
```

Comment:
I'm not entirely sure if I understand the purpose of this useEffect hook is, but it looks like its meant to add the new ingredient to the database. If that's the case, you don't actually need a useEffect hook here. Instead, you can put your fetch request directly in the addIngr function. Hope this helps!